haystack/components/builders/__init__.py

```diff
@@ -2,5 +2,12 @@
 from haystack.components.builders.prompt_builder import PromptBuilder
 from haystack.components.builders.dynamic_prompt_builder import DynamicPromptBuilder
 from haystack.components.builders.dynamic_chat_prompt_builder import DynamicChatPromptBuilder
+from haystack.components.builders.hypothetical_doc_query import HypotheticalDocumentEmbedder
 
-__all__ = ["AnswerBuilder", "PromptBuilder", "DynamicPromptBuilder", "DynamicChatPromptBuilder"]
+__all__ = [
+    "AnswerBuilder",
+    "PromptBuilder",
+    "DynamicPromptBuilder",
+    "DynamicChatPromptBuilder",
+    "HypotheticalDocumentEmbedder",
```

Comment:
I wouldn't put the HypotheticalDocumentEmbedder in the builders but in the embedders because of how the component is used in a pipeline. Given a query, in the end it returns an embedding and it can replace other embedders such as a OpenAITextEmbedder.

Reply:
I can move it to the `embedders`